docs/SemanticARC.md

```diff
@@ -194,7 +244,11 @@ In this case, since the apply's second argument must be @owned, a simple use-def
       return %result : $()
     }
 
-While this may look correct to the naked eye, it is actually incorrect even in SIL today. This is because in SIL today, switch_enum always takes arguments at +1! Yet, in the IR there is no indication of the problem and the code will compile! Now let us update the IR given semantic ARC:
+While this may look correct to the naked eye, it is actually incorrect even in SIL today. This is because in SIL today, switch_enum always takes arguments at +1!
+
+[EE] Hm... no. switch_enum doesn't take it's argument as owned.
```

Comment:
Of course there's a release_value %0 in the function; x has to stay valid for the duration of its scope, and SILGen doesn't do the analysis that would be required to prove that this "if let" is unconditional at the end of that scope.

I think you will want both "forwarding" and "borrowing" versions of switch_enum.  In this case, the optimal code-generation is to forward x's ownership into the switch.  If this appeared in a loop, however, you would not want the representation to force x to be copied (retained) just to perform the switch, since validity can be borrowed from the validity of x.